### ISSUE TITLE
Breaks with python-3.6

### ISSUE THREAD

**Issue by yurivict** [coin-or/CyLP#62]
This error:
```
cylp/cy/CyClpPrimalColumnPivotBase.cpp:5019:19: error: no member named 'f_tstate' in '_frame'
        (*frame)->f_tstate = tstate;
        ~~~~~~~~  ^
```

And also non-ascii characters in README.rst cause this error:
```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "setup.py", line 362, in <module>
    s_README = getBdistFriendlyString(open('README.rst').read())
  File "/usr/local/lib/python3.6/encodings/ascii.py", line 26, in decode
    return codecs.ascii_decode(input, self.errors)[0]
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 343: ordinal not in range(128)
*** Error code 1
```

**Comment by yurivict** [comment]
@tkralphs , maybe you could help solve this?

Thank you!
Yuri.

**Comment by sschnug** [comment]
As far as i know, python3 is not supported at all.

There is a pull-request / fork [here](https://github.com/coin-or/CyLP/pull/28) with a lot of discussion. I don't know if it works with py36, but i think at least <=py3.5 works.

**Comment by yurivict** [comment]
Thanks for this info!

**Comment by godofyoga** [comment]
(wradlib) yogesh@yogesh:~/anaconda$ conda install -c jjhelmus cylp
Solving environment: failed

UnsatisfiableError: The following specifications were found to be in conflict:

    cylp -> python=2.7
    python=3.6
    Use "conda info " to see the dependencies for each package.

I am getting  following error . How can i install cylp for python 3.6?

**Comment by philzook58** [comment]
I was able to eventually get this to install and run on python 3.7. I used the py3 branch mentioned here.
https://github.com/coin-or/CyLP/pull/28
https://github.com/jjhelmus/CyLP

`git checkout py3`

It is somewhat similar to the issue here.
https://github.com/scikit-hep/root_numpy/issues/365
It seems that this is a problem in cython generated code. It needs to be generated with the latest version of cython.

I am on a Mac so I think the setup.py was somehow using a rotten version of cython belonging to my system distro of python.

I was able to get it to compile by getting the latest cython
`pip3 install cython`

Disable cython manually in setup.py by setting USECYTHON=False in line 38

Then go into the cylp/cy folder and run

`python3 -m cython --cplus CyCbcModel.pyx`
`python3 -m cython --cplus CyClpSimplex.pyx`

which were the two problem files for me (Although not in your original question?)

Now `python3 setup.py install` should hopefully work in the top folder.
